### ISSUE TITLE
useNetworkd overwrites gateway if ipv4 and ipv6 are both specified

### ISSUE THREAD

**Issue by peterhoeg** [NixOS/nixpkgs#19620]
## Issue description

Specifying the following configuration breaks the creation of the default gateway:

``` nix
networking = {
  useNetworkd = true;
  defaultGateway = "1.2.3.4";
  defaultGateway6 = "2400:0:0::1";
};
```

Only the ipv6 gateway is configured, not the ipv4 gateway leaving the machine unable to connect.

Removing the ipv6 configuration makes it work again.

I think the following code is to blame:

``` nix
genericNetwork = override: {
  DHCP = override (dhcpStr cfg.useDHCP);
} // optionalAttrs (cfg.defaultGateway != null) {
  gateway = override [ cfg.defaultGateway ];
} // optionalAttrs (cfg.defaultGateway6 != null) {
  gateway = override [ cfg.defaultGateway6 ];
} // optionalAttrs (domains != [ ]) {
  domains = override domains;
};
```
### Steps to reproduce

As above.
## Technical details
- NixOS: 17.03.git.231592aM (Gorilla)
- Nix: nix-env (Nix) 1.11.4
- nixpkgs: 17.03.git.f82c855M

**Comment by groxxda** [comment]
Yes, this code looks broken...
Ugly untested fix:
```
genericNetwork = override:
let gateway = optional (cfg.defaultGateway != null) cfg.defaultGateway 
  ++ optional (cfg.defaultGateway6 != null) cfg.defaultGateway6;
in {
  DHCP = override (dhcpStr cfg.useDHCP);
} // optionalAttrs (gateway != [ ]) {
  gateway = override gateway;
} // optionalAttrs (domains != [ ]) {
  domains = override domains;
};

**Comment by fuzzy-id** [comment]
I just ran into this issue. I fixed it by setting the gateway list manually:

{{{
systemd.network.networks."40-eth0".gateway = ["my-ip6-gateway" "my-ip4-gateway"];
}}}

This works although it's pretty ugly.

**Comment by globin** [comment]
Fixed in 6b8ad8b581576d45996a17822e80e6be281dffd9
